File: Bindings~/java/src/test/java/com/alttester/TestsHelper.java
/*
    Copyright(C) 2026 Altom Consulting
*/

package com.alttester;

public class TestsHelper {
  public static int getAltDriverPort() {
    String port = System.getenv("ALTSERVER_PORT");

    if (port != null && port != "") {
      return Integer.parseInt(port);
    }

    return 13000;
  }

  public static String getAltDriverHost() {
    String host = System.getenv("ALTSERVER_HOST");

    if (host != null && host != "") {
      return host;
    }

    return "127.0.0.1";
  }

  public static String getAltDriverPlatformVersion() {
    String platformVersion = System.getenv("ALTSERVER_PLATFORM_VERSION");
    if (platformVersion != null && !platformVersion.isEmpty()) {
      return platformVersion;
    }
    return "unknown";
  }

  public static AltDriver getAltDriver() {
    return new AltDriver(
        TestsHelper.getAltDriverHost(),
        TestsHelper.getAltDriverPort(),
        true,
        60,
        "__default__",
        "unknown",
        TestsHelper.getAltDriverPlatformVersion(),
        "unknown",
        "unknown");
  }
}

File: Assets/Examples/Test/Editor/Driver/TestsHelper.cs
/*
    Copyright(C) 2025 Altom Consulting

    This program is free software: you can redistribute it and/or modify
    it under the terms of the GNU General Public License as published by
    the Free Software Foundation, either version 3 of the License, or
    (at your option) any later version.

    This program is distributed in the hope that it will be useful,
    but WITHOUT ANY WARRANTY; without even the implied warranty of
    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
    GNU General Public License for more details.

    You should have received a copy of the GNU General Public License
    along with this program. If not, see <https://www.gnu.org/licenses/>.
*/

namespace AltTester.AltTesterSDK.Driver.Tests
{
    public class TestsHelper
    {
        public static int GetAltDriverPort()
        {
            string port = System.Environment.GetEnvironmentVariable("ALTSERVER_PORT");
            if (!string.IsNullOrEmpty(port))
            {
                return int.Parse(port);
            }

            return 13000;
        }

        public static string GetAltDriverHost()
        {
            string host = System.Environment.GetEnvironmentVariable("ALTSERVER_HOST");
            if (!string.IsNullOrEmpty(host))
            {
                return host;
            }

            return "127.0.0.1";
        }

        public static AltDriver GetAltDriver()
        {
            return new AltDriver(host: GetAltDriverHost(), port: GetAltDriverPort(), enableLogging: true);
        }

    }
}
